### DISCUSSION TITLE
欢迎大家参与d2ds书籍以及dslings代码练习相关的讨论

### DISCUSSION THREAD

**Discussion by Sunrisepeak** [Sunrisepeak/d2ds#1] [Announcements]
欢迎大家参与d2ds书籍以及dslings代码练习相关的讨论

**Comment by chen-xuewei** [comment]
Array数组的拷贝赋值函数中，如果dsObj的长度小于或者大于当前数组的长度这种情况怎么处理？
Array & operator=(const Array &dsObj) {
        D2DS_SELF_ASSIGNMENT_CHECKER
        for (int i = 0; i < N; i++) {
            mData_e[i] = dsObj.mData_e[i];
        }
        return *this;
    }
    }

**Reply by Sunrisepeak** [reply]
对于定长数组Array 它的类型是 Array<YourType, 8> 和 Array<YourType, 10> 不是同类型。因为数组的长度即模板参数中的N是类型的一部分

对于变长数组会有长度不一致的问题, 例如Vector。可以参考下面的章节
https://sunrisepeak.github.io/d2ds/chapter_02_vector.html

```cpp
template <typename T, typename Alloc = DefaultAllocator>
class Vector {
public:
    Vector & operator=(const Vector &dsObj) {
        D2DS_SELF_ASSIGNMENT_CHECKER
        this->~Vector();
        mSize_e = dsObj.mSize_e;
        mDataPtr_e = static_cast<T *>(Alloc::allocate(sizeof(T) * mSize_e));
        for (int i = 0; i < mSize_e; i++) {
            mDataPtr_e[i] = dsObj.mDataPtr_e[i];
        }
        return *this;
    }
}
```
